python/ray/serve/_private/replica.py

```diff
@@ -403,15 +408,18 @@ async def is_allocated(self) -> str:
             to PENDING_INITIALIZATION startup state.
 
             Returns:
-                The PID, actor ID, node ID, node IP, and log filepath id of the replica.
+                The PID, actor ID, node ID, node IP, node AZ, and log
+                filepath id of the replica.
             """
 
+            node_id = ray.get_runtime_context().get_node_id()
             return (
                 os.getpid(),
                 ray.get_runtime_context().get_actor_id(),
                 ray.get_runtime_context().get_worker_id(),
-                ray.get_runtime_context().get_node_id(),
+                node_id,
                 ray.util.get_node_ip_address(),
+                self._cluster_node_info_cache.get_node_az(node_id),
```

Comment:
we just need to return node_id and deployment_state should be able to get az based on node_id. It has access to the cluster_node_info_cache.